### DISCUSSION TITLE
Supporting multiple separate virtual environments concurrently

### DISCUSSION THREAD

**Discussion by jchristn** [pythonnet/pythonnet#2397] [Q&A]
Hello,

Is it possible, using pythonnet, to support multiple virtual environments running scripts via pythonnet concurrently?

I'm curious about issues that may arise if I have several different scripts that I want to run, each within their own virtual environment, should some of them start while another is running.

The general process I'm following currently is as follows:

1) Assume script1 (with venv `venv_script1`) and script2 (with venv `venv_script2`)
2) When an event fires indicating either script should be loaded:
a) Enter the virtual environment for that script
b) `Py.Import` the script, run the specified method, and retrieve the result
c) Exit the virtual environment

My worry is that if I'm inside of 2b when another event fires, it's going to create chaos.

Thoughts?  Thanks!

**Comment by filmor** [comment]
This is not possible, Python itself doesn't support this. You'll have to spin up individual processes for the scripts. If someone wants to take this up, #2333 would allow you to at least run multiple "venvs" that use the same Python version.
